selector: method (#8854)
fishtown-analytics/docs.getdbt.com: website/docs/reference/node-selection/methods.md

dbt list --select "semantic_model:*"        # list all semantic models 
dbt list --select "+semantic_model:orders"  # list your semantic model named "orders" and all upstream resources
```
<VersionBlock firstVersion="1.12">

### selector

:::info Beta feature
The `selector` method is a beta feature in <Constant name="core" /> v1.12.
:::

> The `selector` method selects the nodes defined by a named [YAML selector](/reference/node-selection/yaml-selectors) in `selectors.yml`. Use it in `--select` or `--exclude` strings so you can compose a named selector with other [methods](/reference/node-selection/methods), [graph operators](/reference/node-selection/graph-operators), and [set operators](/reference/node-selection/set-operators).

Reviewer: <img width="872" height="394" alt="Image" src="https://github.com/user-attachments/assets/74b54e5e-056a-43c9-9732-ad335da56be9" />

Can we move this callout to the selector section rather than putting it right at top? Or even remove it entirely?

Reviewer: Can we also update this page -> https://docs.getdbt.com/reference/node-selection/yaml-selectors?version=1.12#difference-between---select-and---selector

Now that you can use `--select` with `selector:`, I think this content should be updated. 

We should also actually go through [this page](https://docs.getdbt.com/reference/node-selection/yaml-selectors?version=1.12) and replace references to `--selector` with `--select selector:`? Such as:
> Remember that you can always test out your critiera with dbt ls --selector

Given `--selector` is the legacy command, I don't think we need to keep old references to it in our docs.

Reviewer: <img width="852" height="179" alt="Screenshot 2026-04-07 at 8 05 53 AM" src="https://github.com/user-attachments/assets/d7e7ecfd-f867-493c-8f9b-35c7d741ebfa" />

This example also doesn't make sense to me - this is not an example of someone trying to use `--select` and `--selector` together. I think we can just remove the command example entirely.

> The `selector` method selects the nodes defined by a named [YAML selector](/reference/node-selection/yaml-selectors) in `selectors.yml`. You can use it in `--select` or `--exclude` strings so you can compose a named selector with other [methods](/reference/node-selection/methods), [graph operators](/reference/node-selection/graph-operators), and [set operators](/reference/node-selection/set-operators).

Reviewer 2: ```suggestion
The `selector` method selects the nodes defined by a named [YAML selector](/reference/node-selection/yaml-selectors) in `selectors.yml`. Use it in `--select` or `--exclude` strings so you can compose a named selector with other [methods](/reference/node-selection/methods), [graph operators](/reference/node-selection/graph-operators), and [set operators](/reference/node-selection/set-operators).
```

Reviewer 2: Hiya @luna-bianca 

Just amended this to tighten it a tad

```bash
dbt run --select "selector:my_selector"          # same node set as `dbt run --selector my_selector`
dbt build --select "selector:staging selector:nightly"   # union (space-separated)
dbt build --select "selector:staging,selector:nightly"   # intersection (comma-separated)
dbt build --select "1+selector:staging"                  # graph operators
dbt build --select "selector:staging tag:nightly"        # combine with other methods
dbt run --select "selector:staging" --exclude "selector:exclude_tests"
```

> When you use `--selector` together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, dbt raises a warning when `--selector` is combined with `--select` or `--exclude`.

Reviewer 3: Might be useful to add the type of warning raised which is - `SelectExcludeIgnoredWithSelectorWarning`

Author: Applied [here](https://github.com/dbt-labs/docs.getdbt.com/pull/8854/changes/44f8f86cd361738cfa7d42673b8c289ab6e118c7)

> When you use `--selector` together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, dbt raises the `SelectExcludeIgnoredWithSelectorWarning` warning when `--selector` is combined with `--select` or `--exclude`.

Author: ```suggestion
When you use `--selector` together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, dbt raises the `SelectExcludeIgnoredWithSelectorWarning` warning when `--selector` is combined with `--select` or `--exclude`. If you want to combine a selector with `--select` or `--exclude`, use the `selector:` method instead.
```

> When you use `--selector` together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, dbt raises the `SelectExcludeIgnoredWithSelectorWarning` warning when `--selector` is combined with `--select` or `--exclude`.

Reviewer 2: ```suggestion
When you use `--selector` together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, <Constant name="dbt" /> raises the `SelectExcludeIgnoredWithSelectorWarning` warning when `--selector` is combined with `--select` or `--exclude`.
```

> When you use `--selector` together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, dbt raises `SelectExcludeIgnoredWithSelectorWarning` when `--selector` is combined with `--select` or `--exclude`. If you want to combine a selector with these flags, use the `selector:` method instead.

Author: ```suggestion
When you use the legacy `--selector` flag together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, dbt raises `SelectExcludeIgnoredWithSelectorWarning` when `--selector` is combined with `--select` or `--exclude`. If you want to combine a selector with these flags, use the `selector:` method instead.
```

> When you use `--selector` together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, dbt raises the `SelectExcludeIgnoredWithSelectorWarning` warning when `--selector` is combined with `--select` or `--exclude`. If you want to combine a selector with `--select` or `--exclude`, use the `selector:` method instead.

Author: ```suggestion
When you use `--selector` together with `--select` or `--exclude`, dbt only uses `--selector` for node selection and ignores `--select` and `--exclude`. Starting in <Constant name="core" /> v1.12, dbt raises `SelectExcludeIgnoredWithSelectorWarning` when `--selector` is combined with `--select` or `--exclude`. If you want to combine a selector with these flags, use the `selector:` method instead.
```

> When you use `--select` or `--exclude`, dbt does not apply any default selector. To include a selector in the selection, you must explicitly reference it using the `selector:` method.

Reviewer: But if you have an "unqualified" command, then the default selector will be used https://docs.getdbt.com/reference/node-selection/yaml-selectors?version=1.11#default

Author: Updated [here](https://github.com/dbt-labs/docs.getdbt.com/pull/8854/changes/78a9f28446c73423e5d3ef9fb7ecec09f2c47d81)

> When you use `--select` or `--exclude`, dbt does not apply any default selector. To include a selector in the selection, you must explicitly reference it using the `selector:` method.

Reviewer 2: ```suggestion
When you use `--select` or `--exclude`, Constant name="dbt" /> does not apply any default selector. To include a selector in the selection, you must explicitly reference it using the `selector:` method.
```

Author: Same comment as above

> If selector definitions reference each other in a cycle, dbt raises an error. For more information, refer to [Selector inheritance](/reference/node-selection/yaml-selectors#selector-inheritance).

Reviewer 3: Earlier with `--selector` flag cycling selector inheritance was flagged at compile time itself. So I think we should clarify -

dbt raises a `DbtRecursionError` error at runtime

Author: Applied [here](https://github.com/dbt-labs/docs.getdbt.com/pull/8854/changes/44f8f86cd361738cfa7d42673b8c289ab6e118c7)

> If selector definitions reference each other in a cycle, dbt raises the `DbtRecursionError` error at runtime. For more information, refer to [Selector inheritance](/reference/node-selection/yaml-selectors#selector-inheritance).

Reviewer 2: ```suggestion
If selector definitions reference each other in a cycle, dbt raises the `DbtRecursionError` at runtime. For more information, refer to [Selector inheritance](/reference/node-selection/yaml-selectors#selector-inheritance).
```
